src/main/java/org/mongen/core/models/CollaboratorType.java

```diff
@@ -0,0 +1,71 @@
+package org.mongen.core.models;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import javax.persistence.*;
+import java.io.Serializable;
+import java.util.Date;
+import java.util.List;
+
+@Entity
+@Table(name="collaborator_type")
+public class CollaboratorType implements Serializable{
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	private Long id;
+	private String name;
+
+	@JsonIgnore
+	@Column(updatable=false)
+	private Date created;
+	@JsonIgnore
+	private Date updated;
+
+	@JsonIgnore
+	@OneToMany(mappedBy="type")
+	private List<Collaborator> collaborators;
+
+	public CollaboratorType() {
+	}
+	@PrePersist
+    protected void onCreate(){
+        this.created = new Date();
+    }
+	
+	@PreUpdate
+    protected void onUpdate(){
+        this.updated = new Date();
+    }
+	public Long getId() {
+		return id;
```

Comment:
i guess we can use Lombok to minimise those methods https://projectlombok.org/features/Data
So if we use @Lombok here, we don't need this method 
```
public Long getId() {
 return id;
}
```
we can just call getId() wherever we need, and it will work out of the box

Reply:
Good idea, will try